src/c++/perf_analyzer/test_request_rate_manager.cc

```diff
@@ -889,4 +900,114 @@ TEST_CASE("Request rate - Shared memory infer input calls")
     CHECK(actual_set_shared_memory_calls > 0);
   }
 }
+
+TEST_CASE("request_rate_deadlock")
+{
+  PerfAnalyzerParameters params{};
+  params.max_concurrency = 6;
+  bool is_sequence_model{true};
+  bool some_infer_failures{false};
+
+  const auto& ParameterizeSync{[&]() {
+    SUBCASE("sync")
+    {
+      params.async = false;
+      params.streaming = false;
+    }
+    SUBCASE("aync no streaming")
+    {
+      params.async = true;
+      params.streaming = false;
+    }
+    SUBCASE("async streaming")
+    {
+      params.async = true;
+      params.streaming = true;
+    }
+  }};
+
+  const auto& ParameterizeConcurrency{[&]() {
```

Comment:
This obviously doesn't belong here. Is there something else that does?

Reply:
[params.request_rate_range](https://github.com/triton-inference-server/client/blob/main/src/c%2B%2B/perf_analyzer/command_line_parser.h#L66)?

Reply:
I'm just going to change threads. Rate shouldn't matter much -- due to the delay it will always be behind schedule.